CoreScriptsRoot/Modules/BackpackScript.lua

```diff
@@ -663,6 +719,150 @@ local function MakeSlot(parent, index)
 	return slot
 end
 
+local function AppendNewSlot()
+	local index = #Slots + 1
+
+	local slot = MakeSlot(ScrollingFrame, index)
+
+	return slot, index
+end
+
+local function GetFirstEmptySlot()
+	-- Always returns Slot, SlotIndex
+
+	if LowestEmptySlot ~= nil then
+		return LowestEmptySlot, LowestEmptySlot.Index
+	else
+		return AppendNewSlot()
+	end
+end
+
+local function InsertNewSlot(index)
+	-- Not actually inserting a slot since slots can't move.
+	-- Actually appending a slot and then shifting every relevant tool one slot to the right (up in the array)
+
+	-- Append the new slot
+	local newSlot, newIndex = GetFirstEmptySlot()
+
+	-- Make sure we aren't trying to insert past the length of the backpack
+	if index > newIndex then
+		index = newIndex
+	end
+
+	-- Perform the right-shift.  This is technically a swap operation but since our appended slot is empty it will behave like a shift.
+	for checkIndex = newIndex, index, -1 do
+		local thisSlot = Slots[checkIndex]
+		local nextSlot = Slots[checkIndex+1]
+
+		if nextSlot ~= nil then
+			thisSlot:Swap(nextSlot)
+		end
+	end
+
+	-- Return the "inserted" slot, which is now empty because the appended slot's emptiness was handed down the chain
+	return Slots[index]
+end
+
+local function GetConfiguredToolSlotIndex(toolName)
+	-- Returns the slot index that the user has placed the tool
+	-- If too much time has passed since the player spawned, return nil
+
+	if tick() - CharacterSpawnedAtTime > CONFIGURED_LOADOUT_GRACE_PERIOD then
+		return nil
+	end
+
+	local savedIndex = UserConfiguration[toolName]
+
+	-- Quick check for first tool
+	if next(Slots) == nil then
+		return 1
+	end
+
+	-- Return the last position if it doesn't exist in configuration
+	if savedIndex == nil then
+		if LowestEmptySlot == nil then
+			return #Slots + 1
+		else
+			return LowestEmptySlot.Index
+		end
+	end
+
+	-- Find currently loaded slots
+	local slotIndexes = {}
+
+	for i, slot in pairs(Slots) do
+		if slot.Tool then
+			slotIndexes[slot.Tool.Name] = slot.Index
+		end
+	end
+
+	-- Find closest currently-loaded tools that are below amd above the saved configuration
+	local closestBelow, closestBelowIndex, closestAbove, closestAboveIndex
+
+	for i, otherSlot in pairs(Slots) do
+		local otherTool = otherSlot.Tool
+
+		if otherTool ~= nil then
+			local otherName = otherTool.Name
+			local otherSavedIndex = UserConfiguration[otherName]
+
+			if otherSavedIndex ~= nil then
+				if otherSavedIndex < savedIndex then -- Other tool was saved below this tool
+					if closestBelow == nil or savedIndex - closestBelowIndex > savedIndex - otherSavedIndex then
```

Comment:
> savedIndex - closestBelowIndex > savedIndex - otherSavedIndex

Is mathematically equivalent to 
`-closestBelowIndex > -otherSavedIndex`
which is equal to
`closestBelowIndex < otherSavedIndex`